src/Views/MediaViewer.vala

```diff
@@ -103,10 +103,12 @@ public class Tuba.Views.MediaViewer : Gtk.Widget, Gtk.Buildable, Adw.Swipeable {
 				if (!is_video) return;
 				if (is_done) {
 					#if CLAPPER
+						Clapper.Player player = ((ClapperGtk.Video) child_widget).player;
+
 						if (value) {
-							((ClapperGtk.Video) child_widget).player.play ();
-						} else {
-							((ClapperGtk.Video) child_widget).player.pause ();
+							player.play ();
+						} else if (player.state > Clapper.PlayerState.STOPPED) {
+							player.pause ();
```

Comment:
I am not sure if desired outcome when switching between videos/GIFV is to pause previous one and later resume (like its currently done) or to just stop and play from beginning when selected again?

Reply:
I haven't put much thought into it, but the only thing I can think of in favor of pausing, is that someone might be watching a long video or gif and accidentally swipe to the next attachment, losing the video progress